api/services/sshkeys.go

```diff
@@ -30,17 +30,35 @@ type Request struct {
 	Namespace string
 }
 
-func (s *service) EvaluateKeyHostname(ctx context.Context, key *models.PublicKey, dev models.Device) (bool, error) {
-	if key.Hostname == "" {
-		return true, nil
+func (s *service) EvaluateKeyFilter(ctx context.Context, key *models.PublicKey, dev models.Device) (bool, error) {
+	exist := func(item string, list []string) bool {
+		for _, elem := range list {
+			if elem == item {
+				return true
+			}
+		}
+
+		return false
 	}
 
-	ok, err := regexp.MatchString(key.Hostname, dev.Name)
-	if err != nil {
-		return false, err
+	if key.Filter.Hostname != "" {
+		ok, err := regexp.MatchString(key.Filter.Hostname, dev.Name)
+		if err != nil {
+			return false, err
+		}
+
+		return ok, nil
+	} else if len(key.Filter.Tags) > 0 {
+		for _, tag := range dev.Tags {
+			if exist(tag, key.Filter.Tags) {
```

Comment:
I understood the use of lambda here however it is used only once; I'd put the `for` here.